tests/testTestMethods.py

```diff
@@ -211,6 +213,218 @@ def testAssertWcssNearlyEqualOverBBox(self):
+        mi1 = mi.Factory(mi, True)
+        mask0 = mi0.getMask()
+        mask1 = mi1.getMask()
+        for dval in (1, 3):
+            mask1.getArray()[2, 2] += 1 # getArray avoids "unsupported operand type" failure
+            with self.assertRaises(Exception):
+                self.assertMasksEqual(mask0, mask1)
+            with self.assertRaises(Exception):
+                self.assertMasksEqual(mask1, mask0)
+            with self.assertRaises(Exception):
+                self.assertMaskedImagesEqual(mi0, mi1)
+            with self.assertRaises(Exception):
+                self.assertMaskedImagesEqual(mi1, mi0)
+
+        skipMask = mi.getMask().Factory(mi.getMask(), True)
+        skipMaskArr = skipMask.getArray()
+        skipMaskArr[:] = 0
+        skipMaskArr[2, 2] = 1
+        self.assertMasksEqual(mask0, mask1, skipMask=skipMaskArr)
+        self.assertMasksEqual(mask0, mask1, skipMask=skipMask)
+        self.assertMaskedImagesNearlyEqual(mi0, mi1, skipMask=skipMaskArr, atol=0, rtol=0)
+        self.assertMaskedImagesNearlyEqual(mi0, mi1, skipMask=skipMask, atol=0, rtol=0)
+
+    def testAssertImagesNearlyEqual(self):
+        """Test assertImagesNearlyEqual, assertMasksNearlyEqual and assertMaskedImagesNearlyEqual
+        """
+        width = 10
+        height = 9
+
+        for miType in (afwImage.MaskedImageF, afwImage.MaskedImageD, afwImage.MaskedImageI,
+            afwImage.MaskedImageU):
+            mi = makeMaskedImage(width, height, miType)
+            self.checkMaskedImage(mi)
+
+        for invalidType in (str, self.assertRaises, object):
+            mi = makeMaskedImage(width, height, miType)
+            with self.assertRaises(TypeError):
+                self.assertMasksEqual(mi.getMask(), invalidType)
+            with self.assertRaises(TypeError):
+                self.assertMasksEqual(invalidType, mi.getMask())
+            with self.assertRaises(TypeError):
+                self.assertMasksEqual(mi.getMask(), mi.getMask(), skipMask=invalidType)
+
+            with self.assertRaises(TypeError):
+                self.assertImagesNearlyEqual(mi.getImage(), invalidType)
+            with self.assertRaises(TypeError):
+                self.assertImagesNearlyEqual(invalidType, mi.getImage())
+            with self.assertRaises(TypeError):
+                self.assertImagesNearlyEqual(mi.getImage(), mi.getImage(), skipMask=invalidType)
+
+            with self.assertRaises(TypeError):
+                self.assertMaskedImagesNearlyEqual(mi, invalidType)
+            with self.assertRaises(TypeError):
+                self.assertMaskedImagesNearlyEqual(invalidType, mi)
+            with self.assertRaises(TypeError):
+                self.assertMaskedImagesNearlyEqual(mi, mi, skipMask=invalidType)
+
+            with self.assertRaises(TypeError):
+                self.assertMaskedImagesNearlyEqual(mi.getImage(), mi.getImage())
+
+
+def makeMaskedImage(width, height, imgClass=afwImage.MaskedImageF):
```

Comment:
This name is somewhat confusing given that it is making a ramp image, much like `makeRampMaskedImage`, but with additional nan/inf/-inf values. Would be more obvious if it was called something like `makeRampMaskedImageWithNans` :smile: